Add accessibility guidance for Placeholders (#680)
primer/design: content/guides/accessibility/placeholders.mdx

---
title: Placeholders
description: Placeholders have many accessibility and usability concerns to be aware of.
---

## Overview

> Placeholders are hint text placed inside of an input. They are used to help suggest waht kind of input the should be provided, but have some accessibility considerations to be aware of.

Reviewer: ```suggestion
Placeholders are hint text placed inside of an input. They are used to help suggest what kind of input should be provided, but have some accessibility considerations to be aware of.
```

### Best practices

- [Avoid using placeholders](https://primer.style/components/text-input#accessibility) if at all possible.
- Instead, use a [caption](https://primer.style/ui-patterns/forms/overview#caption) to communicate this information.
- If you must use a placeholder, be sure that the input also has a concise, descriptive label.

## Why?

> Placeholders are typically styled using a light text treatment to visually communicate that they are suggestions, and not pre-filled input. However, this often means they are too light to be color contrast compliant. If the placeholder color treatment is darked to be compliant, it risks being visually interpreted as pre-filled input.

Author: ```suggestion
Placeholders are typically styled using a light text treatment to visually communicate that they are suggestions, and not pre-filled input. However, this often means they are too light to be color contrast compliant. If the placeholder color treatment is darkened to be compliant, it risks being visually interpreted as pre-filled input.
```

> Placeholder length is also contrained by the width of the input it is used in. This means important information may be truncated, especially if the user relies on an increased font size.

Reviewer 2: ```suggestion
Placeholder length is also constrained by the width of the input it is used in. This means important information may be truncated, especially if the user relies on an increased font size.
```
